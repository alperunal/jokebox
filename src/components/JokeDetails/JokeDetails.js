import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const JokeDetails = props => {
  const { joke } = props;
  return (
    <JokeContainer>
      <Info>
        <Setup>{joke.setup}</Setup>
        <Punchline>{joke.punchline}</Punchline>
      </Info>
    </JokeContainer>
  );
};

export const JokeContainer = styled.div`
  display: flex;
  flex-direction: 'row';
  background-color: #ffffff;
`;

export const Info = styled.div`
  width: 100%;
  padding: '50px';
  display: flex;
  flex-direction: column;
  text-align: center;
`;

const Setup = styled.div`
  margin-top: 100px;
  color: #111111;
  font-size: 24px;
`;

const Punchline = styled.div`
  margin-top: 100px;
  font-size: 20px;
  max-height: 150px;
  overflow: auto;
  font-style: italic;
`;

JokeDetails.propTypes = {
  joke: PropTypes.shape({
    id: PropTypes.number,
    type: PropTypes.string,
    setup: PropTypes.string,
    punchline: PropTypes.string,
  }).isRequired,
};

export default JokeDetails;
