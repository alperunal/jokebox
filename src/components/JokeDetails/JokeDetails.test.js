import React from 'react';
import { shallow } from 'enzyme';
import JokeDetails from './JokeDetails';

describe('JokeDetails', () => {
  let wrapper;
  const jokeData = {
    id: 21,
    type: 'general',
    setup: 'A termite walks into a bar and says...',
    punchline: "'Where is the bar tended?'",
  };

  it('renders JokeDetails correctly', () => {
    wrapper = shallow(<JokeDetails joke={jokeData} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.contains(jokeData.setup)).toBe(true);
    expect(wrapper.contains(jokeData.punchline)).toBe(true);
  });
});
