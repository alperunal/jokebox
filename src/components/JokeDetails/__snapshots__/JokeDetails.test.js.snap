// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JokeDetails renders JokeDetails correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      A termite walks into a bar and says...
    </styled.div>
    <styled.div>
      'Where is the bar tended?'
    </styled.div>
  </styled.div>
</styled.div>
`;
